.github/workflows/linting.yml

```diff
@@ -0,0 +1,54 @@
+name: Linting
+
+on: [pull_request]
+
+jobs:
+  linting:
+    name: Linting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Get Node version
+        id: get-node-version
+        run: echo ::set-output name=NODE_VERSION::$(cat .nvmrc)
+
+      - name: Setup Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ steps.get-node-version.outputs.NODE_VERSION }}
+
+      - name: Cache dependencies
+        id: cache-dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/yarn
+            node_modules
+          key: ${{ steps.get-node-version.outputs.NODE_VERSION }}-${{ hashFiles('yarn.lock') }}
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile --prefer-offline
+        env:
+          YARN_CACHE_FOLDER: ~/.cache/yarn
+
+      - name: Get files changed
+        id: get-files-changed
+        uses: trilom/file-changes-action@v1.2.3
```

Comment:
I would look into what this action is comparing against. I imagine it's only comparing to the previous commit.

For example, if we're running linting on a PR, it would miss out on linting changes prior to the last commit.
1. Commit A has set of linting errors A'.
2. Commit B has set of linting errors B'.
3. Commit C has no linting errors.
4. Changes that caused linting errors A' and B' still exist, but are no longer caught because linting passed for files changed between B and C.

Reply:
Ah, I see now that this runs on the `pull_request` event.

> If a PR is made then it will look at all of the files included in the PR. If a push is made then it will compare commits from the SHA github.payload.before to the SHA github.payload.after of the push.

Going to have to think about how to use linting in the main CI... Maybe the annotations belong in the manual review triggers workflow, and the CI workflow doesn't annotate at all?

Reply:
Yeah I had to move to `pull_request` otherwise I would be running into that same issue you mentioned about it only comparing to the previous commit. 

We can definitely do that. It still a in a form of `pull_request` so it should still have the same expected output.
> I think we'll still want the linting to happen in the main CI workflow. If there's an issue, it should be caught and fixed before getting archived, deployed, etc.


The only thing is it would have to moved from CI. Unfortunately I was not able to think of a way to capture only file diff within a push